doc/userguide/capture-hardware/ebpf-xdp.rst

```diff
@@ -0,0 +1,304 @@
+
+::
+
+ XDP or eXpress Data Path provides a high performance, programmable network data path in the Linux kernel as part of the IO Visor Project. XDP provides bare metal packet processing at the lowest point in the software stack which makes it ideal for speed without compromising programmability. Furthermore, new functions can be implemented dynamically with the integrated fast path without kernel modification.
+
+`More info about XDP <https://www.iovisor.org/technology/xdp>`__
+
+Requirements
+------------
+
+- Newer kernel that supports XDP (4.13.10 in our case)
+- RSS symmetric hashing on the NIC ( Intel 82599ES 10-Gigabit/x520/x540 in our case)
+- In tree kernel drivers NIC drivers
+
+Prerequisites
+-------------
+
+This guide has been confirmed on Debian/Ubuntu LTS Linux.
+
+Disable irqbalance
+~~~~~~~~~~~~~~~~~~
+
+::
+
+ systemctl stop irqbalance
+ systemctl disable irqbalance
+
+Kernel
+~~~~~~
+
+Install kernel 4.13.+ and reboot.
+
+Clang
+~~~~~
+
+Make sure you have clang (>3.9) installed on the system  ::
+
+ apt-get install clang
+
+BPF
+~~~
+
+Suricata use libbpf to interact with eBPF and XDP. This library is available
+in the Linux tree. Before Linux 4.16, a patched libbpf library is also needed::
+
+ git clone -b libbpf-xdp-v5  https://github.com/regit/linux.git
+
+If you have a recent enough kernel, you can skip this part.
+
+Now, you can build and install the library ::
+
+ cd linux/tools/lib/bpf/
+ make && sudo make install
+
+ sudo mkdir -p /usr/local/include/bpf/
+ sudo cp *bpf.h /usr/local/include/bpf/
+ sudo ldconfig
+
+
+Compile and install Suricata
+----------------------------
+
+If ever you don't have the source ::
+
+ git clone  https://github.com/OISF/suricata.git
+ cd suricata && git clone https://github.com/OISF/libhtp.git -b 0.5.x
+
+ ./autogen.sh
+
+Then you need to add the ebpf flags to configure ::
+
+ CC=clang ./configure --prefix=/usr/ --sysconfdir=/etc/ --localstatedir=/var/ \
+ --enable-ebpf --enable-ebpf-build
+
+ make clean && make
+ sudo  make install-full
+ sudo ldconfig
+
+Setup bypass
+------------
+
+If you plan to use eBPF or XDP for a kernel/hardware level bypass, you need to do
+the following:
+
+First, enable `bypass` in the `stream` section ::
+
+ stream:
+   bypass: true
+
+If you with you can also bypass encrypted flow by setting no-reassemble to yes
+in the app-layer tls section ::
+
+  app-layer:
+    protocols:
+      tls:
+        enabled: yes
+        detection-ports:
+          dp: 443
+  
+        # Completely stop processing TLS/SSL session after the handshake
+        # completed. If bypass is enabled this will also trigger flow
+        # bypass. If disabled (the default), TLS/SSL session is still
+        # tracked for Heartbleed and other anomalies.
+        no-reassemble: yes
+
+
+Setup eBPF filter
+-----------------
+
+Copy the resulting ebpf fiter as needed ::
+
+ cp src/ebpf/vlan_filter.bpf /etc/suricata/
```

Comment:
I think it would be good to use a separate dir from the start. e.g. /etc/suricata/filter/ or /etc/suricata/bpf/ or something